core/primitives/src/runtime/config_store.rs

```diff
@@ -0,0 +1,103 @@
+use include_dir::{include_dir, Dir};
+
+use crate::runtime::config::RuntimeConfig;
+use crate::types::Gas;
+use crate::types::ProtocolVersion;
+use std::collections::BTreeMap;
+use std::iter::FromIterator;
+use std::ops::Bound;
+use std::sync::Arc;
+
+/// Stores runtime config for each protocol version where it was updated.
+#[derive(Debug)]
+pub struct RuntimeConfigStore {
+    /// Maps protocol version to the config with possibly modified `max_gas_burnt_view` limit.
+    store: BTreeMap<ProtocolVersion, Arc<RuntimeConfig>>,
+}
+
+impl RuntimeConfigStore {
+    /// Constructs a new store.
+    ///
+    /// If `max_gas_burnt_view` is provided, the property in wasm limit
+    /// configuration will be adjusted to given value.
+    pub fn new(max_gas_burnt_view: Option<Gas>) -> Self {
+        let runtime_configs_dir: Dir = include_dir!("../../nearcore/res/runtime_configs");
+        Self {
+            store: BTreeMap::from_iter(runtime_configs_dir.files().iter().map(|file| {
+                let mut config: RuntimeConfig = serde_json::from_slice(file.contents()).unwrap();
+                if let Some(gas) = max_gas_burnt_view {
+                    config.wasm_config.limit_config.max_gas_burnt_view = gas;
+                }
+                (
+                    file.path().file_stem().unwrap().to_str().unwrap().parse().unwrap(),
+                    Arc::new(config),
+                )
```

Comment:
This is somewhat hard to read. Could you introduce intermediate variables to make this more readable? Also I feel like there are too many `unwrap`s here and could potentially be simplified. cc @matklad

Reply:
I agree that current implementation of `RuntimeConfigStore::new` can be confusing, because it has internal logic unrelated to storage creation.

I think that making `RuntimeConfigStore::new` trivial and having a function `RuntimeConfigStore::from_max_gas_burnt_view` resolves it, because we need some function for building configs using custom `max_gas_burnt_view` anyway.

"Apply view-related parameters outside of `NightshadeRuntime`" looks like a good separate goal. Though, at first I would put `RuntimeConfigStore::from_max_gas_burnt_view(max_gas_burnt_view)` to `NightshadeRuntime` and then change the signature of `NightshadeRuntime` and put `RuntimeConfigStore` creation outside, because the latter change would affect lots of places.

Reply:
I agree with what @matklad suggested here. But I am also wondering, why are we setting `max_gas_burnt_view` separately at all? Why can't we just use the value set in `config` and just set `config.max_gas_burnt_view` to be the correct value at the callers of `NightshadeRuntime::new`. Setting this field separately seems very hacky.

And that should be done in a separate PR.

Reply:
Yep, I agree with it. From what I see, `max_gas_burnt_view` is taken from `RuntimeConfig` only due to historical reasons.

Reply:
Removing `max_gas_burnt_view` here: https://github.com/near/nearcore/pull/4716